app/controllers/job_messages_controller.rb

```diff
@@ -0,0 +1,11 @@
+class JobMessagesController < ApplicationController
+  before_filter  :authorize_only_for_admin
+
+  def show
+    @job_message = Rails.cache.fetch(params[:job_id])
```

Comment:
I was under the impression that anything in before_enqueue ran synchronously which would ensure that a jobmessage existed. Were you able to replicate this issue?